# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/IndexedDatabaseInlines.h
# Commit: 7c9ee023c8cc
# Full Hash: 7c9ee023c8cc72aab0269cebeb7a2f02c5281cd1
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-04-01 21:26:59
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefPtr for FileInfo. r=dom-workers-and-storage-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65550
# ==============================================================================

diff -r 7e9d56eb485e -r 7c9ee023c8cc dom/indexedDB/IndexedDatabaseInlines.h
--- a/dom/indexedDB/IndexedDatabaseInlines.h	Wed Apr 01 09:52:54 2020 +0000
+++ b/dom/indexedDB/IndexedDatabaseInlines.h	Wed Apr 01 09:53:01 2020 +0000
@@ -48,7 +48,7 @@
 }
 
 inline StructuredCloneFileParent::StructuredCloneFileParent(
-    FileType aType, RefPtr<indexedDB::FileInfo> aFileInfo)
+    FileType aType, SafeRefPtr<indexedDB::FileInfo> aFileInfo)
     : StructuredCloneFileBase{aType}, mContents{Some(std::move(aFileInfo))} {
   MOZ_ASSERT(**mContents);
   MOZ_COUNT_CTOR(StructuredCloneFileParent);
@@ -75,9 +75,9 @@
   MOZ_COUNT_DTOR(StructuredCloneFileParent);
 }
 
-inline RefPtr<indexedDB::FileInfo> StructuredCloneFileParent::FileInfoPtr()
+inline SafeRefPtr<indexedDB::FileInfo> StructuredCloneFileParent::FileInfoPtr()
     const {
-  return **mContents;
+  return (*mContents)->clonePtr();
 }
 
 inline RefPtr<dom::Blob> StructuredCloneFileChild::BlobPtr() const {